pkg/testutils/leakcheck.go

```diff
@@ -0,0 +1,13 @@
+package testutils
+
+import (
+	"go.uber.org/goleak"
+)
+
+// IgnoreGlogFlushDaemonLeak returns a goleak.Option that ignores the flushDaemon function
+// from the glog package that can cause false positives in leak detection.
+// This is necessary because glog starts a goroutine in the background that may not
+// be stopped when the test finishes, leading to a detected but expected leak.
+func IgnoreGlogFlushDaemonLeak() goleak.Option {
+	return goleak.IgnoreTopFunction("github.com/golang/glog.(*fileSink).flushDaemon")
+}
```

Comment:
add package_test invoking this function to appease code coverage

Reply:
Also, I just realized, we could've done this differently by adding `testutils.CheckGoroutineLeaks` function that would be called from the rest of the codebase, instead of having two dependencies everywhere (goleak and testutils).

Reply:
yeah, we can also do that.